From: Chris Lalancette <clalance@redhat.com>
Date: Fri, 20 Mar 2009 10:21:40 +0100
Subject: [misc] printk: add KERN_CONT
Message-id: 49C36024.5020906@redhat.com
O-Subject: [RHEL5.4 PATCH 1/14]: printk: add KERN_CONT
Bugzilla: 463573
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>
RH-Acked-by: Justin M. Forbes <jforbes@redhat.com>

printk: add the KERN_CONT annotation (which is empty string but via
which checkpatch.pl can notice that the lacking KERN_ level is fine).
This useful for multiple calls of hand-crafted printk output done by
early debug code or similar.

upstream commit 4749252776712e587c7c7619cbf0342b080c7f44

Fixes BZ 463573

diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index fbbdf8f..12a94b6 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -58,6 +58,13 @@ extern const char linux_banner[];
 #define	KERN_INFO	"<6>"	/* informational			*/
 #define	KERN_DEBUG	"<7>"	/* debug-level messages			*/
 
+/*
+ * Annotation for a "continued" line of log printout (only done after a
+ * line that had no enclosing \n). Only to be used by core/arch code
+ * during early bootup (a continued line is not SMP-safe otherwise).
+ */
+#define        KERN_CONT       ""
+
 extern int console_printk[];
 
 #define console_loglevel (console_printk[0])
